crates/sui-core/src/consensus_adapter.rs

```diff
@@ -576,8 +589,18 @@ impl ConsensusAdapter {
         lock: Option<&RwLockReadGuard<ReconfigState>>,
         epoch_store: &Arc<AuthorityPerEpochStore>,
     ) -> SuiResult<JoinHandle<()>> {
-        epoch_store.insert_pending_consensus_transactions(&transaction, lock)?;
-        Ok(self.submit_unchecked(transaction, epoch_store))
+        let transactions = vec![transaction];
+        self.submit_batch(transactions, lock, epoch_store)
+    }
+
+    pub fn submit_batch(
+        self: &Arc<Self>,
+        transactions: Vec<ConsensusTransaction>,
```

Comment:
can be `&[ConsensusTransaction]` to avoid allocating temporary vector (same comment applies below as well)

(if you already tried this and ran into lifetime problems, let me know)

Reply:
Yes there is indeed a lifetime issue.   Inside `submit_unchecked` there is a call to `ConsensusAdapter.submit_and_wait`, which is an async fn.